core/src/main/java/io/questdb/cutlass/http/processors/JsonQueryProcessor.java

```diff
@@ -107,8 +107,7 @@ public void execute0(JsonQueryProcessorState state) throws PeerDisconnectedExcep
         final HttpConnectionContext context = state.getHttpConnectionContext();
         // do not set random for new request to avoid copying random from previous request into next one
         // the only time we need to copy random from state is when we resume request execution
-        sqlExecutionContext.with(context.getCairoSecurityContext(), null, null);
-        sqlExecutionContext.setRequestFd(context.getFd());
+        sqlExecutionContext.with(context.getCairoSecurityContext(), null, null, context.getFd(), context.getSqlExecutionInterruptor());
```

Comment:
our users use questdb as a library and we had mild complaints that SqlExecutionContext keep changing from version to version. Perhaps overloading methods here can go a long way to keep users happy.

Reply:
I'm guilty of doing this as well. I didn't mean to put you on the spot here, but we perhaps should start changing the course of the execution context :)